rust/ql/test/query-tests/unusedentities/unreachable.rs

```diff
@@ -127,33 +147,89 @@ pub fn unreachable_panic() {
     }
 }
 
-pub fn unreachable_match() {
-    match get_a_number() {
-        1 => {
-            return;
-        }
-        _ => {
-            do_something();
-        }
+macro_rules! bail_1 {
+    () => {
+        return Err(String::from("message"))
+    };
+}
+
+macro_rules! bail_2 {
+    ($message:literal) => {
+        return Err(String::from($message))
+    };
+}
+
+pub fn unreachable_bail() -> Result<i32, String> {
+    if cond() {
+        do_something();
+        return Err(String::from("message"));
+        do_something(); // $ Alert[rust/dead-code]
+    }
+
+    if cond() {
+        do_something();
+        bail_1!(); // $ SPURIOUS: Alert[rust/dead-code]
+        do_something(); // $ MISSING: Alert[rust/dead-code]
```

Comment:
This is an issue with the workarounds for `PostOrderTree` and macro nodes.  I'll figure out a way to fix it.

Reply:
I've created an issue to track this, I know what to do but I have other priorities.